packages/platform-core/src/usage-data/get_installation_id.ts

```diff
@@ -0,0 +1,18 @@
+import { userInfo } from 'os';
+import { v5 as uuidV5 } from 'uuid';
+import { CwdPackageJsonReader } from '../cwd_package_json_reader.js';
+
+// eslint-disable-next-line spellcheck/spell-checker
+const AMPLIFY_CLI_UUID_NAMESPACE = 'e7368840-2eb6-4042-99b4-9d6c2a9370e6'; // A random v4 UUID
+/**
+ * Generates a consistent installation Uuid from the cwd + packageJsonName + userName
+ */
+export const getInstallationUuid = (
+  namespace: string = AMPLIFY_CLI_UUID_NAMESPACE
+) => {
+  const packageJsonName = new CwdPackageJsonReader().read().name;
+  const userName = userInfo().username;
+  const modulePath = __dirname;
+
+  return uuidV5(modulePath + packageJsonName + userName, namespace);
```

Comment:
My vote would still be for just `modulePath + userName`. While packageJsonName should be pretty stable, it is subject to change (we have changed our package.json names on occasion)

Reply:
I think in that case it's acceptable for installation id to change. I think it's worse to have two customers sharing the installation id vs one customer to have many. Nevertheless I'm not strong on that so changed it to remove package.json name

Reply:
Maybe for machine id we could use `os.hostname()` for now ? https://nodejs.org/api/os.html#oshostname